code/modules/reagents/reagent_containers.dm

```diff
@@ -80,7 +80,53 @@ var/list/LOGGED_SPLASH_REAGENTS = list(FUEL, THERMITE)
 	return
 
 /obj/item/weapon/reagent_containers/attack(mob/M as mob, mob/user as mob, def_zone)
-	return
+	//If harm intent, splash it on em, else try to feed em it
+
+	if(!M.reagents)
+		return
+
+	if(!is_open_container())
+		to_chat(user, "<span class='warning'>You can't, \the [src] is closed.</span>")//Added this here and elsewhere to prevent drinking, etc. from closed drink containers. - Hinaichigo
+		return
+
+	if(!src.reagents.total_volume)
+		to_chat(user, "<span class='warning'>\The [src] is empty.<span>")
+		return 0
+
+	if(user.a_intent != I_HELP)
+		if(src.reagents)
+			transfer(M, user, splashable_units = -1)
+			playsound(get_turf(M), 'sound/effects/slosh.ogg', 25, 1)
+			return 1
+
+
+	else if(M == user)
+		imbibe(user)
+		return 1
+
+	else if(ishuman(M))
+		user.visible_message("<span class='danger'>[user] attempts to feed [M] \the [src].</span>", "<span class='danger'>You attempt to feed [M] \the [src].</span>")
+
+		var/drink_delay = 30*(amount_per_transfer_from_this/5) //So normal = 1
+		if(!do_mob(user, M, drink_delay))
+			return 1
+
+		user.visible_message("<span class='danger'>[user] feeds [M] \the [src].</span>", "<span class='danger'>You feed [M] \the [src].</span>")
+
+		M.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been fed [src.name] by [user.name] ([user.ckey]) Reagents: [reagentlist(src)]</font>")
```

Comment:
Replace from this down to setting LAssailant with a call to add_attacklogs, it handles all the stuff for you.